crates/types/src/config/common.rs

```diff
@@ -431,16 +439,43 @@ pub enum LogFormat {
 #[builder(default)]
 #[serde(rename_all = "kebab-case")]
 pub struct MetadataStoreClientOptions {
-    /// Metadata store server to bootstrap the node from.
+    /// # Metadata store
+    ///
+    /// Metadata store server kind.
     pub metadata_store_client: MetadataStoreClient,
 
+    /// # Connect timeout
+    ///
+    /// TCP connection timeout for connecting to the metadata store.
+    #[serde_as(as = "serde_with::DisplayFromStr")]
+    #[cfg_attr(feature = "schemars", schemars(with = "String"))]
+    pub metadata_store_connect_timeout: humantime::Duration,
+
     /// # Backoff policy used by the metadata store client
     ///
     /// Backoff policy used by the metadata store client when it encounters concurrent
     /// modifications.
     pub metadata_store_client_backoff_policy: RetryPolicy,
 }
 
+impl CommonClientConnectionOptions for MetadataStoreClientOptions {
+    fn connect_timeout(&self) -> Duration {
+        self.metadata_store_connect_timeout.into()
+    }
+
+    fn http2_keep_alive_interval(&self) -> Option<Duration> {
+        None
+    }
+
+    fn http2_keep_alive_timeout(&self) -> Option<Duration> {
+        None
+    }
```

Comment:
We don't currently expose these for metadata clients, but they could be useful there. It might also be handy to move the retry policy here too, one day.